client/src/app/site/pages/meetings/pages/mediafiles/modules/mediafile-list/components/mediafile-list/mediafile-list.component.ts

```diff
@@ -171,7 +172,13 @@ export class MediafileListComponent extends BaseMeetingListViewComponent<ViewMed
         return false;
     }
 
-    public changeDirectory(directoryId: number | null): void {
+    public async changeDirectory(directoryId: number | null): Promise<void> {
+        const mediafilesSubscribed = await this.modelRequestService.subscriptionGotData(MEDIAFILES_SUBSCRIPTION);
```

Comment:
Shouldn't this wait until data was received? Or why is the timeout below necessary?

Independently, please introduce a common base class to not duplicate the function code. You can probably combine even more functionality in that, I did not look through the rest of the classes.

Reply:
It is possible that the subscription was not opened yet. In that case `subscriptionGotData` returns false. 

I second that but I think we should open another issue for that.